sm_python/python/sm/__init__.py

```diff
@@ -83,3 +83,18 @@ def logFatalNamed(name,message):
 
 def logErrorNamed(name,message):
     logNamed(name, LoggingLevel.Error, message, stackdepth=2)
+
+def setExtendableValueStoreValue(evs, key, value, add_if_missing=True):
+    type2func_string = {
+        bool : "Bool",
+        int : "Int",
+        float : "Double",
+        str : "String"
+    }
+    if evs.hasKey(key):
+        getattr(evs, "set" + type2func_string[value.__class__])(key, value)
+    elif add_if_missing:
+        getattr(evs, "add" + type2func_string[value.__class__])(key, value)
```

Comment:
I'd rather only do the `haKey` test if `add_if_missing`. Otherwise you can simply perform the `set` and let it fail with the usual message if it doesn't work.